GIT_STYLE_GUIDE.md

```diff
@@ -0,0 +1,264 @@
+# Git Style Guide
+
+This Git Style Guide directly borrows, extends, and adapts from the following guides:
+
+* [Git Style Guide by agis-](https://github.com/agis-/git-style-guide)
+* [Git branch naming conventions by lovingly](http://www.guyroutledge.co.uk/blog/git-branch-naming-conventions/)
+* [Branching by digitalijhelms](https://gist.github.com/digitaljhelms/4287848)
+
+# Table of contents
+
+1. [Branches](#branches)
+2. [Commits](#commits)
+  1. [Messages](#messages)
+3. [Merging and Pull Requests](#merging)
+4. [Issues](#issues)
+5. [Stuff to check later](#misc)
+
+# Branches
+
+<table>
+  <thead>
+    <tr>
+      <th>Instance</th>
+      <th>Branch</th>
+      <th>Description, Instructions, Notes</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>Master</td>
+      <td>master</td>
+      <td>Accepts merges from Develop and Hotfixes</td>
+    </tr>
+    <tr>
+      <td>Develop</td>
+      <td>develop</td>
+      <td>Accepts merges from us/Issues and Hotfixes</td>
+    </tr>
+    <tr>
+      <td>Task</td>
+      <td>task/short-desc</td>
+      <td>Always branch off HEAD of develop, it referes to an iteration task described on Trello.</td>
+    </tr>
+    <tr>
+      <td>Hotfix</td>
+      <td>hotfix/short-desc</td>
+      <td>Always branch off Master, receives utmost priority for development, testing and code reviews. After merging and fixing Master in production it's changes get merged into Develop</td>
+    </tr>
+    <tr>
+      <td>Issue</td>
+      <td>issue/{{issue-id}}-short-desc</td>
+      <td>Never branches off Master, it may branch from any other branch as necessary but primarily branches off  develop.</td>
+    </tr>
+    <tr>
+      <td>Docs</td>
+      <td>docs/short-desc</td>
+      <td>Branches off develop, and will contain documentation that does not refer to any specific piece of code i.e. this guide.</td>
+    </tr>
+  </tbody>
+</table>
+
+* Choose *short* and *descriptive* names:
+
+  ```shell
+  # good
+  $ git checkout -b task/oauth-migration
+
+  # bad - too vague
+  $ git checkout -b task/login_fix
+  ```
+
+* Always follow the folder structure outlined above:
+
+  ```shell
+  # good
+  $ git checkout -b bug-fix/utf-enconding
```

Comment:
I know this is a very stupid detail from y part. But it ways "follow folder structure from structure outlined above. bug-fix is not outlined above. I think there should be a category  that is outline above like "issue"

Just because some people in both teams (knowing them) might get confused.